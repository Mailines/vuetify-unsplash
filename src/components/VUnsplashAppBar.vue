<template>
  <v-app-bar v-bind="appBarProps" fixed app>
    <v-hover v-slot:default="{ hover }">
      <v-btn
        :class="{ rotateIconVunsplash: hover }"
        id="close-dialog-app-bar-btn-v-unsplash"
        class="ml-4 mr-1"
        icon
        @click="closeDialog"
      >
        <v-icon>mdi-close</v-icon>
      </v-btn>
    </v-hover>
    <v-toolbar-title>{{ appBarTitle }}</v-toolbar-title>
    <v-spacer></v-spacer>
    <v-btn
      class="mr-6"
      id="save-app-bar-btn-v-unsplash"
      large
      :color="saveButtonOutlined ? '' : '#00000000'"
      @click="saveSelection"
      :disabled="!selectionSaveable"
      :outlined="saveButtonOutlined"
      >{{ saveButtonLabel
      }}<v-icon class="ml-4" right>{{ saveButtonIcon }}</v-icon></v-btn
    >
  </v-app-bar>
</template>

<script>
export default {
  props: {
    appBarProps: {
      type: Object,
      required: true
    },
    appBarTitle: {
      type: String,
      required: true
    },
    selectionSaveable: {
      type: Boolean,
      required: true
    },
    saveButtonLabel: {
      type: String,
      required: true
    },
    saveButtonIcon: {
      type: String,
      required: true
    },
    saveButtonOutlined: {
      type: Boolean,
      required: true
    }
  },
  methods: {
    closeDialog() {
      this.$emit("close");
    },
    saveSelection() {
      this.$emit("save");
    }
  }
};
</script>
<style scoped>
.rotateIconVunsplash {
  animation-timing-function: ease-in-out;
  transform: rotate(-180deg);
}
</style>
