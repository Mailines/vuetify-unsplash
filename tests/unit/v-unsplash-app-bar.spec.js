import VUnsplashAppBar from "../../src/components/VUnsplashAppBar";
import { mount, createLocalVue } from "@vue/test-utils";
import Vuetify from "vuetify";
import Vue from "vue";

describe("VUnsplashAppBar.vue", () => {
  let localVue;
  let wrapper;
  let vuetify;

  beforeEach(() => {
    localVue = createLocalVue();
    vuetify = new Vuetify();
    wrapper = mount(VUnsplashAppBar, {
      localVue,
      vuetify,
      propsData: {
        appBarProps: {
          dark: true,
          color: "primary"
        },
        appBarTitle: "Choose an image",
        selectionSaveable: false,
        saveButtonLabel: "Save",
        saveButtonIcon: "mdi-content-save",
        saveButtonOutlined: true
      }
    });
  });
  it("renders and match snapshot", async () => {
    expect(wrapper.exists()).toBe(true);
    // With jest we can create snapshot files of the HTML output
    expect(wrapper.html()).toMatchSnapshot();
  });
  it("disabled the save button correctly", () => {
    expect(wrapper.find("#save-app-bar-btn-v-unsplash").props("disabled")).toBe(
      true
    );
  });
  it("enables save button correctly when selectionSaveable=true", async () => {
    wrapper.setProps({
      selectionSaveable: true
    });
    await Vue.nextTick();
    expect(wrapper.find("#save-app-bar-btn-v-unsplash").props("disabled")).toBe(
      false
    );
  });
  it("emits close event when close button is clicked", () => {
    wrapper.find("#close-dialog-app-bar-btn-v-unsplash").trigger("click");
    expect(wrapper.emitted().close).toBeTruthy();
  });
  it("emits save event when save button is clicked", async () => {
    wrapper.setProps({
      selectionSaveable: true
    });
    await Vue.nextTick();
    wrapper.find("#save-app-bar-btn-v-unsplash").trigger("click");
    expect(wrapper.emitted().save).toBeTruthy();
  });
});
