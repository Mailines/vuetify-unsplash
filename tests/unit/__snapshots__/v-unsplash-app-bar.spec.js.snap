// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VUnsplashAppBar.vue renders and match snapshot 1`] = `
<header class="v-sheet theme--dark v-toolbar v-app-bar v-app-bar--fixed primary" style="height: 64px; margin-top: 0px; transform: translateY(0px); left: 0px; right: 0px;">
  <div class="v-toolbar__content" style="height: 64px;"><button type="button" class="ml-4 mr-1 v-btn v-btn--flat v-btn--icon v-btn--round theme--dark v-size--default" id="close-dialog-app-bar-btn-v-unsplash"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-close theme--dark"></i></span></button>
    <div class="v-toolbar__title">Choose an image</div>
    <div class="spacer"></div> <button type="button" disabled="disabled" class="mr-6 v-btn v-btn--depressed v-btn--disabled v-btn--flat v-btn--outlined theme--dark v-size--large" id="save-app-bar-btn-v-unsplash"><span class="v-btn__content">Save<i aria-hidden="true" class="v-icon notranslate ml-4 v-icon--right mdi mdi-content-save theme--dark"></i></span></button>
  </div>
</header>
`;
